### PR TITLE
Add types to invoke.Runner.run

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

**Review by AlexWaygood — CHANGES_REQUESTED** [review]
Thanks! A few points below.

As well as the things I've mentioned below:
- We generally prefer to use protocols for I/O types, rather than concrete classes. I haven't done a deep investigation into the source code, but it looks like `_typeshed.SupportsWrite` and might work well for `in_stream` and `out_stream`. Could you maybe take a look?

**Comment by kasium** [comment]
So, invoke uses sys.stdout/stderr by default for streams and they have TextIO as their type I would like to use them

File: stubs/invoke/invoke/runners.pyi [hunk 1]
```diff
@@ -9,7 +12,30 @@ class Runner:
...
+        command: str,
+        asynchronous: bool = ...,
+        disown: bool = ...,
+        dry: bool = ...,
+        echo: bool = ...,
+        echo_format: str = ...,
+        echo_stdin: bool | None = ...,
+        encoding: str = ...,
+        err_stream=...,
+        env: dict[str, str] = ...,
```
[the remark this answers is not in the document]
Comment (kasium) [reply stubs/invoke/invoke/runners.pyi:27]: done

**Comment by kasium** [comment]
Can you please check the overload? In short:

- if disown is True-> return None (asynchronous is ignored)
- if asynchronous is True and disown is False or default -> return Promise
- if asynchronous is False or default and disown is False or default  -> return Result

File: stubs/invoke/invoke/runners.pyi [hunk 2]
```diff
@@ -1,5 +1,11 @@
-from typing import Any
-from typing_extensions import Literal
+from collections.abc import Iterable, Mapping
+from io import TextIO
+from typing import Any, overload
```
Comment (AlexWaygood) [inline stubs/invoke/invoke/runners.pyi:3]:
```suggestion
from typing import Any, TextIO, overload
```

**Comment by AlexWaygood** [comment]
> Can you please check the overload? In short:
> 
> * if disown is True-> return None (asynchronous is ignored)
> * if asynchronous is True and disown is False or default -> return Promise
> * if asynchronous is False or default and disown is False or default  -> return Result

I've taken the liberty of pushing to your branch to fix various things. Could you take another look and check that everything looks okay from your end?

**Comment by kasium** [comment]
wow, thanks a lot! Looks great

**Review by AlexWaygood — APPROVED** [review]
Great, thank you!

**Pull request merged**